File: 2574. Left and Right Sum Differences.cpp
class Solution {
public:

    vector<int> leftRightDifference(vector<int>& numbers) {

        int arraySize = numbers.size();

        // Calculate total sum of all elements
        int remainingRightSum = 0;

        for (int currentNumber : numbers) {
            remainingRightSum += currentNumber;
        }

        // Sum of elements to the left of current index
        int currentLeftSum = 0;

        vector<int> differences(arraySize);

        for (int index = 0; index < arraySize; index++) {

            // Remove current element from right sum
            remainingRightSum -= numbers[index];

            // Store absolute difference
            differences[index] =
                abs(currentLeftSum - remainingRightSum);

            // Add current element to left sum
            currentLeftSum += numbers[index];
        }

        return differences;
    }
};

File: 2574. Left and Right Sum Differences.py
class Solution:
    def leftRightDifference(self, numbers: List[int]) -> List[int]:

        # Sum of all elements (initial right sum)
        remaining_right_sum = sum(numbers)

        # Sum of elements on the left side
        current_left_sum = 0

        # Result array
        differences = [0] * len(numbers)

        for index in range(len(numbers)):

            # Remove current element from right sum
            remaining_right_sum -= numbers[index]

            # Store absolute difference
            differences[index] = abs(
                current_left_sum - remaining_right_sum
            )

            # Add current element to left sum
            current_left_sum += numbers[index]

        return differences
